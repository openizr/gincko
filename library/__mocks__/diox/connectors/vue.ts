/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/**
 * diox/connectors/vue mock.
 */
const connect = vi.fn(() => vi.fn(() => ({
  steps: [{ id: 'start' }, { id: 'end' }],
  loading: process.env.LOADING === 'true',
})));

export default connect;
