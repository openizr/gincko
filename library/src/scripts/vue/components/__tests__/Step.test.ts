/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { Component } from 'vue';
import { mount } from '@vue/test-utils';
import Step from 'scripts/vue/components/Step.vue';

type ComponentApi = {
  attrs: {
    id?: string;
  };
};

jest.mock('scripts/vue/components/Field', () => ({
  render(createElement: (tag: string, api: ComponentApi) => Component): Component {
    return createElement('div', {
      attrs: {
        id: 'Field',
      },
    });
  },
  mounted(): void {
    (this as unknown as { $emit: (eventName: string) => void; }).$emit('userAction');
  },
}));
jest.mock('sonar-ui/vue', () => ({
  markdown: (value: string): string => value,
  buildClass: (...values: string[]): string => values.join(' '),
}));

describe('vue/components/Step', () => {
  const onUserAction = jest.fn();
  const customComponents = {};
  const i18n = (): string => 'Test';

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('active step', () => {
    const wrapper = mount(Step, {
      propsData: {
        i18n,
        id: 'step',
        status: 'success',
        isActive: true,
        index: 0,
        fields: [{
          id: 'Message',
          type: 'Unknown',
          status: 'initial',
          options: {},
        }],
        customComponents,
      },
      listeners: {
        userAction: onUserAction,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('inactive step', () => {
    const wrapper = mount(Step, {
      propsData: {
        i18n,
        id: 'step',
        status: 'success',
        isActive: false,
        index: 0,
        fields: [{
          id: 'Message',
          type: 'Unknown',
          status: 'initial',
          options: {},
        }],
      },
      listeners: {
        userAction: onUserAction,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
