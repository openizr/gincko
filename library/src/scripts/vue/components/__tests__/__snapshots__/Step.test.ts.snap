// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue/components/Step active step 1`] = `
"<div class=\\"form-step success,step,active\\">
  <div class=\\"form-step__fields\\">
    <div id=\\"Message\\" active=\\"true\\" type=\\"Unknown\\" status=\\"initial\\" options=\\"[object Object]\\" custom-components=\\"[object Object]\\"></div>
  </div>
</div>"
`;

exports[`vue/components/Step inactive step 1`] = `
"<div class=\\"form-step success,step,\\">
  <div class=\\"form-step__fields\\">
    <div id=\\"Message\\" type=\\"Unknown\\" status=\\"initial\\" options=\\"[object Object]\\" custom-components=\\"[object Object]\\"></div>
  </div>
</div>"
`;
